### PR TITLE
Add nerdctl create command

### PR DESCRIPTION
**Pull request by Junnplus**
Signed-off-by: ye.sijun <junnplus@gmail.com>

### CODE REVIEW

**Comment by Junnplus** [comment]
```
# nerdctl create nginx -- echo foo
fc31dbadd94fee65a97ebb56d6043439ab8593bd9b7cdb4298bdfc7fb15466eb
# nerdctl ps -a
CONTAINER ID    IMAGE                             COMMAND                   CREATED           STATUS     PORTS    NAMES
fc31dbadd94f    docker.io/library/nginx:latest    "/docker-entrypoint.…"    16 seconds ago    Created
# nerdctl start fc31dbadd94f
DEBU[0000] stateDir: /run/user/501/containerd-rootless
DEBU[0000] rootless parent main: executing "/usr/bin/nsenter" with [-r/ -w/Users/jun/Documents/workspace/nerdctl --preserve-credentials -m -n -U -t 2186 -F nerdctl start fc31dbadd94f]
DEBU[0000] failed to delete old task                     error="task must be stopped before deletion: created: failed precondition"
FATA[0000] task fc31dbadd94fee65a97ebb56d6043439ab8593bd9b7cdb4298bdfc7fb15466eb already exists: unknown
```
The start command implemented now will delete the task, but the task in the created state cannot be deleted. 
https://github.com/containerd/containerd/blob/v1.6.0-beta.4/task.go#L308-L313

**Comment by AkihiroSuda** [comment]
Why do you need to remove the task when it is in `Created` state?

Also, please add tests

**Comment by Junnplus** [comment]
> Why do you need to remove the task when it is in Created state?

I did not remove the task, but the current implementation of the start command will delete it.
https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/start.go#L93-L97

**Comment by Junnplus** [comment]
@AkihiroSuda Sorry, my faul. 

I found that the state created by the container is `Unknown`, so I created a task to fix `Unknown`, but it was wrong, seems to be related to https://github.com/containerd/nerdctl/pull/647

File: cmd/nerdctl/create_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+func TestCreate(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+
+	base.Cmd("create", "--name", tID, testutil.CommonImage, "echo", "foo").AssertOK()
+	defer base.Cmd("rm", "-f", tID).Run()
+	base.Cmd("ps", "-a").AssertOutContains("Created")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/create_test.go:32]: Please also test that the container can be started with "start"

Comment (Junnplus) [reply cmd/nerdctl/create_test.go:32]: @AkihiroSuda Yes, I am still waiting for https://github.com/containerd/nerdctl/pull/647 to merge

File: cmd/nerdctl/create.go [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+	if err != nil {
+		return nil, "", nil, err
+	}
+	if env := strutil.DedupeStrSlice(env); len(env) > 0 {
+		opts = append(opts, oci.WithEnv(env))
+	}
+
+	if flagT {
+		opts = append(opts, oci.WithTTY)
+	}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/create.go:321]: Seems already set in L249

File: cmd/nerdctl/create.go [hunk 3]
```diff
@@ -0,0 +1,443 @@
+/*
```
Comment (AkihiroSuda) [inline cmd/nerdctl/create.go:1]: Maybe we can put the entire `create.go` into `run.go`. These two commands are not decouplable, and it is hard to find what function belongs to `create.go` and what to `run.go`

Comment (Junnplus) [reply cmd/nerdctl/create.go:1]: SGTM, I am also confused about this question : )

**Comment by AkihiroSuda** [comment]
```
=== RUN   TestIssue108
    main_linux_test.go:33: assertion failed: expression is false: res.ExitCode == exitCode: ghcr.io/stargz-containers/alpine:3.13-org: resolving      |--------------------------------------| 
        elapsed: 0.1 s                             total:   0.0 B (0.0 B/s)                                         
        ghcr.io/stargz-containers/alpine:3.13-org: resolving      |--------------------------------------| 
        elapsed: 0.2 s                             total:   0.0 B (0.0 B/s)                                         
        ghcr.io/stargz-containers/alpine:3.13-org: resolving      |--------------------------------------| 
        elapsed: 0.3 s                             total:   0.0 B (0.0 B/s)                                         
        ghcr.io/stargz-containers/alpine:3.13-org:                                     resolved       |++++++++++++++++++++++++++++++++++++++| 
        index-sha256:ec14c7992a97fc11425907e908340c6c3d6ff602f5f13d899e6b7027c9b4133a: downloading    |--------------------------------------|    0.0 B/1.6 KiB 
        elapsed: 0.4 s                                                                 total:   0.0 B (0.0 B/s)                                         
        ghcr.io/stargz-containers/alpine:3.13-org:                                        resolved       |++++++++++++++++++++++++++++++++++++++| 
        index-sha256:ec14c7992a97fc11425907e908340c6c3d6ff602f5f13d899e6b7027c9b4133a:    done           |++++++++++++++++++++++++++++++++++++++| 
        manifest-sha256:e103c1b4bf019dc290bcc7aca538dc2bf7a9d0fc836e186f5fa34945c5168310: downloading    |--------------------------------------|    0.0 B/528.0 B 
        elapsed: 0.5 s                                                                    total:  1.6 Ki (3.2 KiB/s)                                       
        ghcr.io/stargz-containers/alpine:3.13-org:                                        resolved       |++++++++++++++++++++++++++++++++++++++| 
        index-sha256:ec14c7992a97fc11425907e908340c6c3d6ff602f5f13d899e6b7027c9b4133a:    done           |++++++++++++++++++++++++++++++++++++++| 
        manifest-sha256:e103c1b4bf019dc290bcc7aca538dc2bf7a9d0fc836e186f5fa34945c5168310: done           |++++++++++++++++++++++++++++++++++++++| 
        config-sha256:49f356fa4513676c5e22e3a8404aad6c7262cc7aaed15341458265320786c58c:   done           |++++++++++++++++++++++++++++++++++++++| 
        elapsed: 0.6 s                                                                    total:  1.6 Ki (2.7 KiB/s)                                       
        FATA[0000] OCI runtime create failed: cannot use console socket if runc will not detach or allocate tty: unknown 
        
--- FAIL: TestIssue108 (0.72s)
```

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**